[Curl] Replace CURLOPT_PROTOCOLS with CURLOPT_PROTOCOLS_STR

kshukuwa opened the pull request:
#### 4039058b91cf478f3a733fb0cc371472870342e3
<pre>
[Curl] Replace CURLOPT_PROTOCOLS with CURLOPT_PROTOCOLS_STR
<a href="https://bugs.webkit.org/show_bug.cgi?id=256068">https://bugs.webkit.org/show_bug.cgi?id=256068</a>

Reviewed by Fujii Hironori.

CURLOPT_PROTOCOLS option was deprecated since 7.85.0.
And, it was recommended to use CURLOPT_PROTOCOLS_STR instead.

* Source/WebCore/platform/network/curl/CurlContext.cpp:
(WebCore::CurlHandle::enableAllowedProtocols):
* Source/cmake/OptionsPlayStation.cmake:
* Source/cmake/OptionsWin.cmake:

Canonical link: <a href="https://commits.webkit.org/263482@main">https://commits.webkit.org/263482@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d74a9970ac6cd80f7412844d1668e8e054182b08

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/4768 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/4886 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/5050 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/6275 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/4895 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/4760 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/5040 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/4854 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/5134 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/4847 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/4925 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/4265 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/6287 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/2413 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/4260 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/9232 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/3955 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/4275 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/4334 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/5905 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc-arm64~~](https://ews-build.webkit.org/#/builders/12/builds/4532 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/4742 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/3857 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/4863 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/4254 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/1197 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/8311 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/4988 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/544 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/4612 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/1309 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

fujii commented on Source/WebCore/platform/network/curl/CurlContext.cpp:
```diff
@@ -551,13 +551,13 @@ void CurlHandle::enableAcceptEncoding()
-    static const long allowedProtocols = CURLPROTO_FILE |
 #if ENABLE(FTPDIR)
-        CURLPROTO_FTP | CURLPROTO_FTPS |
+    static constexpr const char* allowedProtocols = "file,ftp,ftps,http,https";
```
I think we don't need both `constexpr` and `const`.

fujii replied:
And, I think 'static' is unnecessary here.

fujii replied:
I think 'auto' is enough.
> auto allowedProtocols = ...;

kshukuwa replied:
Fixed

kshukuwa committed:
[Curl] Replace CURLOPT_PROTOCOLS with CURLOPT_PROTOCOLS_STR
https://bugs.webkit.org/show_bug.cgi?id=256068

Reviewed by Fujii Hironori.

CURLOPT_PROTOCOLS option was deprecated since 7.85.0.
And, it was recommended to use CURLOPT_PROTOCOLS_STR instead.

* Source/WebCore/platform/network/curl/CurlContext.cpp:
(WebCore::CurlHandle::enableAllowedProtocols):
* Source/cmake/OptionsPlayStation.cmake:
* Source/cmake/OptionsWin.cmake:

Canonical link: https://commits.webkit.org/263482@main
